### PR TITLE
Update database query syntax

### PR DESCRIPTION
**Pull request by stnguyen90**
## What does this PR do?

Update Appwrite to use the [new utopia-php/database library with the Queries V2](https://github.com/utopia-php/database/pull/163).

## Test Plan

Automated tests.

image

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -248,31 +248,37 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
...
+        $filterQueries = [];
+
+        if (!empty($search)) {
+            $filterQueries[] = Query::search('search', $search);
+        }
+
+        $queries = [];
+        $queries[] = Query::limit($limit);
+        $queries[] = Query::offset($offset);
+        $queries[] = $orderType === 'ASC' ? Query::orderAsc('') : Query::orderDesc('');
```
Comment (christyjacob4) [inline app/controllers/api/databases.php:266]: where does the query validation take place?

Comment (stnguyen90) [reply app/controllers/api/databases.php:266]: This part doesn't need validation. For List Documents, there's a Query Validator used.

File: app/controllers/general.php [hunk 2]
```diff
@@ -90,15 +90,15 @@
                 if (!empty($envDomain) && $envDomain !== 'localhost') {
                     $mainDomain = $envDomain;
                 } else {
-                    $domainDocument = $dbForConsole->findOne('domains', [], 0, ['_id'], ['ASC']);
+                    $domainDocument = $dbForConsole->findOne('domains', [Query::orderAsc('_id')]);
```
Comment (christyjacob4) [inline app/controllers/general.php:93]: Don't we need to include the offset here ? or is the default offset 0 ? 
I see that we have explicitly used `Query::offset(0)` in some places

Screenshot 2022-08-13 at 2 22 42 PM
 
 Shall we choose one of the patterns and stay consistent ? 
 
 
 Either use the default offset 0 param everywhere or explicitly specify it with `Query::offset(0)`

File: composer.json [hunk 3]
```diff
@@ -44,13 +44,13 @@
         "appwrite/php-runtimes": "0.10.*",
         "utopia-php/framework": "0.20.*",
         "utopia-php/logger": "0.3.*",
-        "utopia-php/abuse": "dev-origin/timestamp-to-datetime as 0.7.2",
+        "utopia-php/abuse": "dev-feat-update-database-queries",
```
Comment (eldadfux) [inline composer.json:47]: Reminder to update all versions.

File: src/Appwrite/Utopia/Database/Validator/Queries.php [hunk 4]
```diff
@@ -10,17 +10,22 @@ class Queries extends ValidatorQueries
     /**
      * Expression constructor
```
Comment (eldadfux) [inline src/Appwrite/Utopia/Database/Validator/Queries.php:11]: Lets add a docblock to explain what we're doing here.

**Review by christyjacob4 — APPROVED** [review]

**Review by eldadfux — COMMENTED** [review]
Minor comments, PR looks great.

File: composer.json (hunk 3 above)
Comment (stnguyen90) [reply composer.json:47]: Ya, will do after the following are merged and release published:

1. https://github.com/utopia-php/database/pull/163
2. https://github.com/utopia-php/abuse/pull/23
3. https://github.com/utopia-php/audit/pull/25

**Pull request merged**